src/app/features/charts/components/median-barchart/median-barchart.component.ts

```diff
@@ -0,0 +1,260 @@
+  implements OnChanges, AfterViewInit, OnDestroy
+{
+  private chartInitialized = false;
+  private tooltipInitialized = false;
+  private _data: MedianExpression[] = [];
+  private chart!: d3.Selection<any, unknown, null, undefined>;
+  private tooltip!: d3.Selection<any, unknown, null, undefined>;
+  private MEANINGFUL_EXPRESSION_THRESHOLD = Math.log2(5);
+  private maxValueY = -1;
+
+  get data() {
+    return this._data;
+  }
+  @Input() set data(data: MedianExpression[]) {
+    this._data = data
+      .filter((el) => el.medianlogcpm && el.medianlogcpm > 0)
+      .sort((a, b) => a.tissue.localeCompare(b.tissue));
+    this.maxValueY = Math.max(
+      this.MEANINGFUL_EXPRESSION_THRESHOLD,
+      d3.max(this._data, (d) => d.medianlogcpm) || 0
+    );
+  }
+
+  @Input() xAxisLabel = '';
+  @Input() yAxisLabel = 'LOG2 CPM';
+
+  @ViewChild('chart') chartRef: ElementRef<SVGElement> = {} as ElementRef;
+  @ViewChild('tooltip') tooltipRef: ElementRef<HTMLElement> = {} as ElementRef;
+
+  dimension: any;
+  group: any;
+
+  constructor(private helperService: HelperService) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (
+      (changes._data && !changes._data.firstChange) ||
+      (changes.xAxisLabel && !changes.xAxisLabel.firstChange) ||
+      (changes.yAxisLabel && !changes.yAxisLabel.firstChange)
+    ) {
+      if (this._data.length === 0) {
+        this.clearChart();
+        this.hideChart();
+      } else {
+        this.clearChart();
+        this.showChart();
+        this.createChart();
+      }
+    }
+  }
+
+  ngAfterViewInit(): void {
+    if (this._data.length === 0) this.hideChart();
+    else this.createChart();
+  }
+
+  ngOnDestroy(): void {
+    this.destroyChart();
+  }
+
+  clearChart() {
+    const svg = d3.select(this.chartRef.nativeElement);
+    svg.selectAll('*').remove();
+  }
+
+  hideChart() {
+    const svg = d3.select(this.chartRef.nativeElement);
+    svg.style('display', 'none');
+  }
+
+  showChart() {
+    const svg = d3.select(this.chartRef.nativeElement);
+    svg.style('display', 'block');
+  }
+
+  destroyChart() {
+    if (this.chartInitialized) this.chart.remove();
+    if (this.tooltipInitialized) this.tooltip.remove();
+  }
+
+  showTooltip(text: string, x: number, y: number): void {
+    this.tooltip = d3
+      .select(this.tooltipRef.nativeElement)
+      .style('left', `${x}px`)
+      .style('top', `${y}px`)
+      .style('display', 'block')
+      .html(text);
+    this.tooltipInitialized = true;
+  }
+
+  hideTooltip() {
+    if (this.tooltipInitialized) {
+      this.tooltip.style('display', 'none');
+    }
+  }
+
+  getBarCenterX(tissue: string, xScale: d3.ScaleBand<string>): number {
+    return (xScale(tissue) || 0) + xScale.bandwidth() / 2;
+  }
+
+  // get the current width allotted to this chart or default
+  getChartBoundingWidth(): number {
+    return (
+      d3.select(this.chartRef.nativeElement).node()?.getBoundingClientRect()
+        .width || 500
+    );
+  }
```

Comment:
Get the current width allotted to this chart, so that the width will update as the window is resized

Reply:
As a suggestion, we can define 500 as a constant at the top of the code.

Reply:
I think you might have moved a lot of the variables out for the scorebarchart component to the top.  You could consider standardizing on that approach if you'd like.

Reply:
Sure thing, done in the next commit